Add build documentation (#123)
KULeuven-MICAS/snitch_cluster: docs/tutorial/build_system.md

# Review of Architecture

> Previously we discussed the [Accelerator Design](./accelerator_design.md), the[CSR Manager Design](./csrman_design.md) and the [Streamer Design](./streamer_design.md). These are the main components that are tightly coupled together in a shell before connecting to the system. The figure below shows the system of interest again but with more labels:

Reviewer: ```suggestion
Previously we discussed the [Accelerator Design](./accelerator_design.md), the [CSR Manager Design](./csrman_design.md) and the [Streamer Design](./streamer_design.md). These are the main components that are tightly coupled together in a shell before connecting to the system. The figure below shows the system of interest again but with more labels:
```

### What About the Naming of the Generated Files?

The names of the generated files are dependent on the configuration file you set. If you recall in the `snax_acc_cfg`, there is a `snax_acc_name` parameter: 

```hjson
snax_acc_cfg: {
    snax_acc_name: "snax_alu"
    snax_tcdm_ports: 16,
    snax_num_rw_csr: 3,
    snax_num_ro_csr: 2,
    snax_streamer_cfg: {$ref: "#/snax_alu_streamer_template" }
},
```
This is prefixed at the beginning of each generated file. Moreover, it also sets the generated directory for each accelerator you add. Such that we have:

```yml
- target: snax-alu
    files:
      # Level 0
      # ... Add some other files here
      - target/snitch_cluster/generated/${snax_acc_name}/${snax_acc_name}_csrman_CsrManager.sv
      - target/snitch_cluster/generated/${snax_acc_name}/${snax_acc_name}_streamer_StreamerTop.sv
      # Level 1
      # ... Add some other files here
      - target/snitch_cluster/generated/${snax_acc_name}/${snax_acc_name}_streamer_wrapper.sv
      # Level 2
      - target/snitch_cluster/generated/${snax_acc_name}/${snax_acc_name}_wrapper.sv
```

> Where each `${snax_acc_name}` is substituted with the parameter name you specified.

Reviewer: Are these to be substituted by the human? Otherwise I would put:
```suggestion
Where you should substitute `${snax_acc_name}`  with the parameter name you specified.
```

Author: Yes, they are human substituted 😅
